webui/react/src/utils/data.ts

```diff
@@ -24,3 +26,45 @@ export const clone = (data: any, deep = true): any => {
   if (isSet(data)) return new Set(data);
   return deep ? JSON.parse(JSON.stringify(data)) : { ...data };
 };
+
+export const stringTimeSorter = (a: string, b: string): number => {
+  const aTime = new Date(a).getTime();
+  const bTime = new Date(b).getTime();
+  return aTime - bTime;
+};
+
+export const alphanumericSorter = (a: string, b: string): number => {
+  return a.localeCompare(b, 'en', { numeric: true });
+};
+
+const runStateSortValues: Record<RunState, number> = {
+  [RunState.Active]: 0,
+  [RunState.Paused]: 1,
+  [RunState.StoppingError]: 2,
+  [RunState.Errored]: 3,
+  [RunState.StoppingCompleted]: 4,
+  [RunState.Completed]: 5,
+  [RunState.StoppingCanceled]: 6,
+  [RunState.Canceled]: 7,
+  [RunState.Deleted]: 7,
+};
+
+const commandStateSortValues: Record<CommandState, number> = {
+  [CommandState.Pending]: 0,
+  [CommandState.Assigned]: 1,
+  [CommandState.Pulling]: 2,
+  [CommandState.Starting]: 3,
+  [CommandState.Running]: 4,
+  [CommandState.Terminating]: 5,
+  [CommandState.Terminated]: 6,
+};
+
+export const stateSorter = (a: State, b: State): number => {
+  // FIXME this is O(n) we can do it in constant time.
+  // What is the right typescript way of doing it?
+  const aValue = Object.values(RunState).includes(a as RunState) ?
+    runStateSortValues[a as RunState] : commandStateSortValues[a as CommandState];
+  const bValue = Object.values(RunState).includes(b as RunState) ?
+    runStateSortValues[b as RunState] : commandStateSortValues[b as CommandState];
+  return aValue - bValue;
+};
```

Comment:
suggestion: You probably wrote this before knowing Tasks List didn't show experiments. With it separated out, this function can be split into one for `RunState` and one for `CommandState`.

misc: Not really an issue because there isn't a collision between `RunState` and `CommandState` values, but if they happen to have the same values in the future, this might produce some funky result.